ui-tests/cypress/integration/pages/widgets-ootb.spec.js

```diff
@@ -0,0 +1,1096 @@
+        const deleteWidgetFromPage = (widgetCode) => {
+          cy.widgetInstanceController(THE_PAGE.code)
+            .then(controller => controller.deleteWidget(widgetCode));
+        };
+        if (Array.isArray(widgetToRemoveFromPage)) {
+          widgetToRemoveFromPage.forEach((widgetCode) => {
+            deleteWidgetFromPage(widgetCode);
+          });
+        } else {
+          deleteWidgetFromPage(widgetToRemoveFromPage);
+        }
+        cy.pagesController()
+          .then(controller => controller.setPageStatus(THE_PAGE.code, 'published'));
+      }
+    });
+    cy.get('@widgetToDelete').then((widgetToDelete) => {
+      if (widgetToDelete !== null) {
+        cy.widgetsController()
+          .then(controller => controller.deleteWidget(SAMPLE_DUPE_WIDGET_CODE));
+      }
+    });
+    cy.get('@widgetToRevert').then((widgetToRevert) => {
+      if (widgetToRevert !== null) {
+        const revertWidget = (widget) => {
+          cy.widgetsController(widget.code)
+            .then(controller => controller.getWidget())
+            .then(({ controller, formData }) => controller.putWidget({ ...formData, ...widget }));
+        };
+        if (Array.isArray(widgetToRevert)) {
+          widgetToRevert.forEach((widget) => {
+            revertWidget(widget);
+          });
+        } else {
+          revertWidget(widgetToRevert);
+        }
+      }
+    });
+    cy.get('@recentContentsToUnpublish').then((contentCounts) => {
+      if (contentCounts !== null && contentCounts > 0) {
+        cy.contentsController()
+          .then(controller => controller.getContentList())
+          .then(({ controller, response }) => {
+            response.body.payload
+              .slice(0, contentCounts).map(content => content.id)
+              .forEach(contentId => controller.updateStatus(contentId, 'draft'));
+          });
+      }
+    });
+    cy.get('@recentContentsToDelete').then((contentCounts) => {
+      if (contentCounts !== null && contentCounts > 0) {
+        cy.contentsController()
+          .then(controller => controller.getContentList())
+          .then(({ controller, response }) => {
+            response.body.payload
+              .slice(0, contentCounts).map(content => content.id)
+              .forEach(contentId => controller.deleteContent(contentId));
+          });
+      }
+    });
+    cy.kcLogout();
+  });
+
+  after(() => {
+    cy.kcLogin('admin').as('tokens');
+    cy.pagesController()
+      .then(controller => {
+        controller.setPageStatus(THE_PAGE.code, 'draft');
+        controller.deletePage(THE_PAGE.code);
+      });
+    cy.kcLogout();
+  });
+
+  const selectPageFromSidebar = (pageOpen = THE_PAGE) => {
+    const currentPageContent = currentPage.getContent();
+    currentPageContent.clickSidebarTab(1);
+    cy.wait(3000);
+    currentPageContent.selectPageFromSidebarPageTreeTable(pageOpen.code);
+    currentPageContent.clickSidebarTab(0);
+  };
+
+  describe('CMS Content Widget', () => {
```

Comment:
The tests in this describe depend on the first test which add the widget to the page, hence they're going to fail if the first one is not execute or if it failed; you should remove the widget at the end of the first test and add it, via API, at the beginning of each other test